ComponentKit/Core/CKComponent.mm

```diff
@@ -275,21 +275,11 @@ + (id)initialState
   return nil;
 }
 
-- (void)updateState:(id (^)(id))updateBlock
-{
-  [self _updateState:updateBlock tryAsynchronousUpdate:NO];
-}
-
-- (void)updateStateWithExpensiveReflow:(id (^)(id))updateBlock
-{
-  [self _updateState:updateBlock tryAsynchronousUpdate:YES];
-}
-
-- (void)_updateState:(id (^)(id))updateBlock tryAsynchronousUpdate:(BOOL)tryAsynchronousUpdate
+- (void)updateState:(id (^)(id))updateBlock mode:(CKUpdateMode)mode
```

Comment:
Yeah, this is a super interesting question. I've discussed it with @modocache, @ocrickard, and many others (including a Facebook-internal post). There is absolutely no consensus on what to do.

Having two separate methods is a straightforward option; e.g. `updateState:` (async) and `synchronouslyUpdateState:` (sync). But now we have bifurcating implementations and it's not entirely obvious to the caller *how* the two are different.

I don't like the idea of exposing a new semantic concept like "priority." This is just another indirect mapping that the caller has to understand. So, mark me down against that one.

In the end, I don't like the idea of forcing everyone to think about it at every point either, but at least the mode has exact symmetry with the updates that you give to a ComponentHostingView or a transactional data source. They behave identically, which makes it entirely obvious what is happening.